numba/targets/arrayobj.py

```diff
@@ -1667,6 +1667,23 @@ def np_reshape_impl(a, shape):
     return np_reshape_impl
 
 
+def _is_nonelike(ty):
+    return (ty is None) or isinstance(ty, types.NoneType)
```

Comment:
This function is also defined on arraymath.py. Maybe both definitions should be moved to a file common to both.

https://github.com/numba/numba/blob/60d2bddd0a9ddf84cf900a70e4b7e17bcbe5dc87/numba/targets/arraymath.py#L3665-L3666

Reply:
Perhaps move it to `numba.numpy_support` for now? There's a load of utility functions there already.